src/dodal/common/udc_directory_provider.py

```diff
@@ -2,24 +2,23 @@
 
 from ophyd_async.core import DirectoryInfo
 
-_directory_info: DirectoryInfo
+from dodal.common.types import UpdatingDirectoryProvider
 
 
-def set_directory(directory: Path):
-    """
-    Set the current base directory of the UDC DirectoryProvider.
-    Files will be stored in the 'panda' subdirectory.
-    """
-    global _directory_info
-    _directory_info = DirectoryInfo(
-        root=directory, resource_dir=Path("panda"), prefix="", suffix=""
-    )
+class UDCDirectoryProvider(UpdatingDirectoryProvider):
+    resource_dir = Path("panda")
 
+    def __init__(self, directory: Path | None = None):
+        self._directory_info = (
+            DirectoryInfo(root=directory, resource_dir=self.resource_dir)
+            if directory
+            else None
```

Comment:
Could: Log a warning if we try and initialise this without specifying a directory